whisper-fine-tuning-event/run_eval_whisper_streaming.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+
+from transformers import pipeline
+from datasets import load_dataset, Audio
+import evaluate
+
+wer_metric = evaluate.load("wer")
+
+
+def is_target_text_in_range(ref):
+    if ref.strip() == "ignore time segment in scoring":
+        return False
+    else:
+        return ref.strip() != ""
+
+
+def get_text(sample):
+    if "text" in sample:
+        return sample["text"]
+    elif "sentence" in sample:
+        return sample["sentence"]
+    elif "normalized_text" in sample:
+        return sample["normalized_text"]
+    elif "transcript" in sample:
+        return sample["transcript"]
+    else:
+        raise ValueError(f"Sample: {sample.keys()} has no transcript.")
+
+
+def data(dataset):
+    for i, item in enumerate(dataset):
+        yield {**item["audio"], "reference": item["norm_text"]}
+
+
+def main(args):
+    batch_size = args.batch_size
+    whisper_asr = pipeline(
+        "automatic-speech-recognition", model=args.model_id, device=args.device
+    )
+    whisper_asr.model.config.max_length = 128
+
+    whisper_norm = whisper_asr.tokenizer._normalize
+
+    def normalise(batch):
+        batch["norm_text"] = whisper_norm(get_text(batch))
+        return batch
+
+    dataset = load_dataset(
+        args.dataset, args.config, split=args.split, streaming=True, use_auth_token=True
```

Comment:
We should allow for non-streaming mode. Participants really want this for smaller datasets:
```suggestion
        args.dataset, args.config, split=args.split, streaming=args.streaming, use_auth_token=True
```

Reply:
Done! Thx